data-products/src/main/scala/org/sunbird/analytics/exhaust/OnDemandDruidExhaustJob.scala

```diff
@@ -0,0 +1,280 @@
+                      threshold: Option[Long], metric: Option[String], descending: Option[String], intervalSlider: Int)
+case class Metrics(metric : String, label : String, druidQuery : DruidQuery)
+case class ReportConfig(id: String, queryType: String, dateRange: QueryDateRange, metrics: List[Metrics], labels: LinkedHashMap[String, String],
+                        output: List[mutable.Map[String,AnyRef]], storageKey: Option[String] = Option(AppConf.getConfig("storage.key.config")),
+                        storageSecret: Option[String] = Option(AppConf.getConfig("storage.secret.config")))
+case class ReportConfigImmutable(id: String, queryType: String, dateRange: QueryDateRange, metrics: List[Metrics], labels: LinkedHashMap[String, String],
+                                 output: List[OutputConfig],mergeConfig: Option[ReportMergeConfig] = None,
+                                 storageKey: Option[String] = Option(AppConf.getConfig("storage.key.config")),
+                                 storageSecret: Option[String] = Option(AppConf.getConfig("storage.secret.config")))
+case class OnDemandDruidResponse(file: List[String], status: String, statusMsg: String, execTime: Long)
+case class FinalMetrics(totalRequests: Option[Int], failedRequests: Option[Int], successRequests: Option[Int])
+
+object OnDemandDruidExhaustJob extends optional.Application with BaseReportsJob with Serializable with OnDemandExhaustJob{
+  override def getClassName() : String = "org.sunbird.analytics.exhaust.OnDemandDruidExhaustJob"
+  val jobName: String = "OnDemandDruidExhaustJob"
+  def name(): String = "OnDemandDruidExhaustJob"
+  // $COVERAGE-OFF$ Disabling scoverage for main and execute method
+  def main(config: String)(implicit sc: Option[SparkContext] = None, fc: Option[FrameworkContext] = None) {
+    JobLogger.init("OnDemandDruidExhaustJob")
+    JobLogger.start("OnDemandDruidExhaustJob Started executing", Option(Map("config" -> config, "model" -> name)))
+    implicit val jobConfig = JSONUtils.deserialize[JobConfig](config)
+    implicit val spark: SparkSession = openSparkSession(jobConfig)
+    implicit val sc: SparkContext = spark.sparkContext
+    implicit val sqlContext = new SQLContext(sc)
+    JobContext.parallelization = CommonUtil.getParallelization(jobConfig)
+
+    implicit val frameworkContext: FrameworkContext = getReportingFrameworkContext()
+    implicit val conf = spark.sparkContext.hadoopConfiguration
+    execute()
+    // $COVERAGE-ON$ Disabling scoverage for main and execute method
+  }
+
+  def validateEncryptionRequest(request: JobRequest): Boolean = {
+    if (request.encryption_key.nonEmpty) true else false;
+  }
+
+  def validateRequest(request: JobRequest): Boolean = {
+    if (request.request_data != null) true else false
+  }
+
+  def getRequests(jobId :String)(implicit spark: SparkSession, fc: FrameworkContext): Array[JobRequest] = {
```

Comment:
getRequests method already available in OnDemandExhaustJob - why is this implemented again here?

Reply:
We Fixed this.